crates/ra_syntax/src/validation/field_expr.rs

```diff
@@ -0,0 +1,14 @@
+use crate::{ast,
+            SyntaxError,
+            SyntaxErrorKind::*,
+};
+use ra_parser::SyntaxKind;
+
+pub(crate) fn validate_field_expr_node(node: &ast::FieldExpr, errors: &mut Vec<SyntaxError>) {
+    let mut c = node.syntax.children();
+    if let Some(n) = c.find(|n| n.kind() == SyntaxKind::TUPLE_INDEX) {
+        if n.text().chunks().flat_map(str::chars).any(|c| c < '0' || c > '9') {

```

Comment:
:+1: 

index is a token, and, for tokens, you can get a `&SmolStr` and a `&str` directly.